### DISCUSSION TITLE
Help debugging tsc paths resolution

### DISCUSSION THREAD

**Discussion by sbalay** [swc-project/swc#7593] [Q&A] [answered]
Hi!

I'm trying swc in my typescript project where we have some custom paths configuration. The config I came up with doesn't seem to work because I'm getting errors while trying to resolve paths that have custom mappings.

Rather than asking for help with config setup, I tried to debug and see what swc was doing under the hood and ended up in [this file](https://github.com/swc-project/swc/blob/ed9a4ae5bc73b54641fc3f60843eb96c8cff6145/crates/swc_ecma_loader/src/resolvers/tsc.rs)

I wanted to add some logs there, so I cloned the project, edited the file, run `yarn build:dev` and then replaced the swc binary in node_modules of my project: `cp ~/projects/swc/swc.darwin-arm64.node ~/projects/my-project/node_modules/@swc/core-darwin-arm64/swc.darwin-arm64.node`

I assumed that I would get some more logs, but I don't see anything when running `yarn swc some-file-of-my-project.ts`.

What would be the proper way of debugging this? Are there any docs around using dev builds?

**Comment by kwonoj** [comment, marked as answer]
It's due to bindings are using pinned, published version from registry instead of path-referencing to the repo's workspace to avoid unexpected version conflict when we publish. 

To make things work with local changes, you need to apply patch to those dependencies to the local path instead. CI have some workfing reference example https://github.com/swc-project/swc/blob/ed9a4ae5bc73b54641fc3f60843eb96c8cff6145/.github/workflows/CI.yml#L170-L175 could be resued for the most of cases.
